velox/exec/tests/TableWriteTest.cpp

```diff
@@ -2262,6 +2264,114 @@ TEST_P(AllTableWriterTest, tableWriteOutputCheck) {
+  for (int i = 0; i < partitionedBy_.size(); ++i) {
+    groupingKeys.emplace_back(std::make_shared<core::FieldAccessTypedExpr>(
+        partitionTypes_.at(i), partitionedBy_.at(i)));
+  }
+
+  // aggregation node
+  core::TypedExprPtr inputField =
+      std::make_shared<const core::FieldAccessTypedExpr>(SMALLINT(), "c2");
+  auto minCallExpr = std::make_shared<const core::CallTypedExpr>(
+      SMALLINT(), std::vector<core::TypedExprPtr>{inputField}, "min");
+  auto maxCallExpr = std::make_shared<const core::CallTypedExpr>(
+      SMALLINT(), std::vector<core::TypedExprPtr>{inputField}, "max");
+  auto distinctCountCallExpr = std::make_shared<const core::CallTypedExpr>(
+      VARCHAR(),
+      std::vector<core::TypedExprPtr>{inputField},
+      "approx_distinct");
+
+  core::TypedExprPtr inputField2 =
+      std::make_shared<const core::FieldAccessTypedExpr>(VARCHAR(), "c5");
+  auto maxDataSizeCallExpr = std::make_shared<const core::CallTypedExpr>(
+      BIGINT(),
+      std::vector<core::TypedExprPtr>{inputField2},
+      "max_data_size_for_stats");
+  auto sumDataSinzeCallExpr = std::make_shared<const core::CallTypedExpr>(
+      BIGINT(),
+      std::vector<core::TypedExprPtr>{inputField2},
+      "sum_data_size_for_stats");
+
+  std::vector<std::string> aggregateNames = {
+      "min",
+      "max",
+      "approx_distinct",
+      "max_data_size_for_stats",
+      "sum_data_size_for_stats"};
+  std::vector<core::AggregationNode::Aggregate> aggregates = {
+      core::AggregationNode::Aggregate{minCallExpr, nullptr, {}, {}},
+      core::AggregationNode::Aggregate{maxCallExpr, nullptr, {}, {}},
+      core::AggregationNode::Aggregate{distinctCountCallExpr, nullptr, {}, {}},
+      core::AggregationNode::Aggregate{maxDataSizeCallExpr, nullptr, {}, {}},
+      core::AggregationNode::Aggregate{sumDataSinzeCallExpr, nullptr, {}, {}},
+  };
+  auto aggregationNode = std::make_shared<core::AggregationNode>(
+      core::PlanNodeId(),
+      core::AggregationNode::Step::kPartial,
+      groupingKeys,
+      std::vector<core::FieldAccessTypedExprPtr>{},
+      aggregateNames,
+      aggregates,
+      false, // ignoreNullKeys
+      PlanBuilder().values({input}).planNode());
+
+  auto plan = PlanBuilder()
+                  .values({input})
+                  .tableWrite(
+                      rowType_,
+                      rowType_->names(),
+                      aggregationNode,
+                      std::make_shared<core::InsertTableHandle>(
+                          kHiveConnectorId,
+                          makeHiveInsertTableHandle(
+                              rowType_->names(),
+                              rowType_->children(),
+                              partitionedBy_,
+                              nullptr,
+                              makeLocationHandle(outputDirectory->path))),
+                      false,
+                      CommitStrategy::kNoCommit)
+                  .planNode();
+
```

Comment:
Can you comment why columnStatsStartIndex is set to 3 + partitionedBy_.size()? Thanks!